terraform/modules/cloud/aws/deploy/fleet/templates/user_data.bash

```diff
@@ -4,11 +4,17 @@
 # set -x
 exec > >(tee /tmp/user-data.log|logger -t user-data ) 2>&1
 
-export INFRASTRUCTURE_ANSIBLE_VAULT_PASSWORD=`aws --region ${region} secretsmanager get-secret-value --secret-id ansible_vault_password --output text --query 'SecretString'`
-
 git clone -b ${bootstrap_version} --single-branch https://github.com/aeternity/infrastructure.git /infrastructure
 cd /infrastructure/ansible
 
+# Temporary fix/workaround for non-executable vault install
+chmod +x /usr/bin/vault
+
+# Authenticate the instance to CSM
+PKCS7=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/pkcs7 | tr -d '\n')
+export VAULT_ADDR=${vault_addr}
+export VAULT_TOKEN=$(vault write -field=token auth/aws/login pkcs7=$PKCS7 role=${vault_role})
```

Comment:
I assume this `auth/aws/login` is the same as documented at https://www.vaultproject.io/api/auth/aws/index.html#login but from CLI.